sale_project_fixed_price_task_completed_invoicing/models/project_task.py

```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class ProjectTask(models.Model):
+    _inherit = 'project.task'
+
+    invoiceable = fields.Boolean(
+        string='Invoiceable',
+    )
+    fixed_price = fields.Boolean(
+        string='Fixed Price',
+    )
+
+    @api.multi
+    def toggle_invoiceable(self):
+        for task in self:
+            # We dont' want to modify when the related SOLine is invoiced
+            if not task.sale_line_id or task.sale_line_id.state in ('done'):
```

Comment:
`in ('done', ) `

missing training comma. Probably 'cancel' should be handled in the same way.